### ISSUE TITLE
No joystick support when installing from flatpak!

### ISSUE THREAD

**Issue by Isakku** [flathub/net.pcsx2.PCSX2#2]
Well well, I'm using Linux Mint KDE 18.3 and I installed the flatpak version of PCSX2 after testing the PPA version (which performed quite bad, sound was crap no matter what I did, Flatpak works as intended however, except for joysticks which don't work!)
I'm dual booting this machine with windows 8.1 and I can use PCSX2 on windows just fine, but I wanted to use it on Linux so I searched (and to my surprise, although several flatpak packages are already available at the Mint software manager, PCSX2 wasn't) so I followed the instructions here

https://forums.pcsx2.net/Thread-Any-plans-for-Appimage-or-Flatpak-support

> 
> flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
> flatpak install --arch=i386 --from https://flathub.org/repo/appstream/net.pcsx2.PCSX2.flatpakref

That got PCSX2 running, but after trying it I noticed it is completely isolated, I had to put iso images of the games within my home folder because PCSX2 was sandboxed, completely restricted to access anything, not even anything within the /MEDIA folder, thus any other drives. Same thing with joysticks, they were completely unavailable, as if they didn't exist!
On the other hand, this hardware is a laptop, so it has a buit-in accelerometer used by the hdapsd daemon to protect the HDD. Said accelerometer is recognized by linux as another joystick (js0) and I really hope PCSX2 will be flexible enough to let me choose my joystick (js1) as the FIRST (player1) joystick and leave js0 alone, because right now Wine doesn't allow me.

Thanks a lot for your time and effort!

**Comment by TingPing** [comment]
It works fine here, just ensure the joystick is connected *before* starting the flatpak.

**Comment by Isakku** [comment]
Hmm... so that was the trick. I wonder why it didn't work before? I thought I had tried connecting, disconnecting, closing PCSX2 and opening it again... but now that you mentioned it it works fine!
Oh, but I sitll can't acess anything within the /media folder!
On a side note, I noticed a slight stutter in some of the sounds (not all but some).
Still it's working much better than the PPA | downloaded from website versions, the sound was just plain noise!
However it  still bothers me that I don't remember those sound stutters when playing on Windows.
Tweaking sound options helps a bit but not much.

**Comment by TingPing** [comment]
> Oh, but I sitll can't acess anything within the /media folder!

You can run this: `flatpak override --user --filesystem=/media:ro net.pcsx2.PCSX2` then restart the app.

**Comment by Isakku** [comment]
Thanks a lot! That did it! I'm sitll started about the joystick issue, I was sure I had it connected when I tested. Maybe I had to reboot my machine before retrying?
About the sound issue, hmm, well It only happens to one game, I guess I will test it again on windows before reporting an issue, but again perhaps I should report it anyways because other PCSX2 versions (non flatpak) have very very bad sound (I can't call it actual sound, it's just noise and crackle).
